<HTML>
<HEAD>
<TITLE>Lecture notes: 6. Beyond the genome</TITLE>
</HEAD>
<BODY BGCOLOR="WHITE">

<A HREF="http://www.sbc.su.se/">Stockholm Bioinformatics Center, SBC</A>
<BR>
<A HREF="index.html">Lecture notes:  Molecular Bioinformatics 2001,
Uppsala University</A>

<P ALIGN="RIGHT">
Lecture 26 Jan 2001
Per Kraulis
</P>

<H2 ALIGN="CENTER">
<A HREF="genomes.html">The genomes</A>
</H2>

<H1 ALIGN="CENTER">6. Beyond the genome</H1>

<P>
The recent complete genome projects, and the results in the form of
analysis and comparison, have driven home the idea that <STRONG>the 
completeness of the data gives an entirely new dimension to biological 
science</STRONG>. New types of analysis can be made and new kinds of 
experiments can designed which simply could not be done before.
</P>

<P>
The realization that the completeness of a data set can be extremely 
useful has made the biological scientific community look into the
possibilities of obtaining other such complete data sets. It has even
spawned a number of new buzzwords, some of which are less elegant than
others:
</P>

<TABLE BORDER CELLPADDING=2>

<TR>
<TH></TH>
<TH>Description</TH>
<TH>Time dependence</TH>
<TH>Manipulation in the cell</TH>
<TH>Measurement in the cell</TH>
</TR>

<TR>
<TH>Genome</TH>
<TD>The complete DNA sequence of the organism.</TD>
<TD>First approximation: none.</TD>
<TD>Genetic manipulation; site-directed mutagenesis, genetic screens.</TD>
<TD>Routine. High accuracy and completeness.</TD>
</TR>

<TR>
<TH>Transcriptome</TH>
<TD>
The complete set of mRNA transcripts in a cell or tissue of the
organism.
</TD>
<TD>Reflects the state of gene expression.</TD>
<TD>Indirectly via genetic manipulation.</TD>
<TD>
Routine, or nearly so. cDNA microarrays, SAGE, Northern blots.
Quantification inaccurate. Approximate completeness.
</TD> 
</TR>

<TR>
<TH>Proteome</TH>
<TD>
The complete set of proteins present in a cell or tissue, including
variants due to covalent modifications. Sometimes also the
associations with other molecular components (complexes).
</TD>
<TD>Responsive to all manner of influences throughout the life cycle.</TD>
<TD>
Indirectly through genetic manipulations. Small molecules as
inhibitors of enzymes.
</TD>
<TD>
Not routine. 2D-gels with mass spectrometry. Fusions with reporter
proteins (GFP). Yeast two-hybrid system. Quantification very
difficult. Completeness far away, and very difficult.
</TD>
</TR>

<TR>
<TH>Metabolome</TH>
<TD>
All metabolites and their concentrations. Defined as organic small
molecules, not protein, RNA or DNA.
</TD>
<TD>Responsive to all manner of influences throughout the life cycle.</TD>
<TD>Difficult.</TD>
<TD>
Not routine. NMR, mass spectrometry, a few other special methods for
specific molecules. Quantification and resolution
difficult. Completeness far away, but maybe reachable.
</TD>
</TR>

</TABLE>

<P>
Of these four kinds of datasets, only the genome and transcriptome can
be routinely measured today. The proteome and metabolome are much more
difficult, and require expertise and well-behaved systems, if at all
possible. There is a severe lack of good experimental strategies to
solve these problems. The interest in developing new methods is
correspondingly great.
</P>

<P>
The future development of biology will to a large extent be based on
new contributions to the characterization and analysis of these data
sets. There is also a vast potential for new applications and new
clever experimental designs that utilize the already established
techniques. And finally, the challenge of synthesizing an overall
picture out of the mass of data is there for bioinformatics to respond to.
</P>

<HR>
Copyright &copy; 2001
Per Kraulis
$Date: 2001/01/24 10:19:52 $
</BODY>
</HTML>
