<HTML>
<HEAD>
<TITLE>Lecture notes: 4. Analysing a genome</TITLE>
</HEAD>
<BODY BGCOLOR="WHITE">

<A HREF="http://www.sbc.su.se/">Stockholm Bioinformatics Center, SBC</A>
<BR>
<A HREF="index.html">Lecture notes:  Molecular Bioinformatics 2001,
Uppsala University</A>

<P ALIGN="RIGHT">
Lecture 26 Jan 2001
Per Kraulis
</P>

<H2 ALIGN="CENTER">
<A HREF="genomes.html">The genomes</A>
</H2>

<H1 ALIGN="CENTER">4. Analysing a genome</H1>

<P>
So, what does one do with a complete genome? After all, a sequenced
genome consists only of so many bases in a defined order. Analysis is
obviously necessary in order to obtain biologically interesting
information. 
</P>

<P>
The analysis of a genome covers many different aspects. Here follows a
list of the most common ones, but it is clear that entirely novel ways
of analysing a complete genome can be invented. The potential for
interesting discoveries in the complete genomes is great; we have
probably just scratched the surface so far.
</P>

<UL>

<LI> Define the location of genes (coding sequences, regulatory
regions): gene prediction (identification).
  <UL>
  <LI> Gene prediction <I>ab initio</I> using software based on rules
  and patterns. Find Open Reading Frames (ORFs), with additional
  criteria for good start sequence for a gene. This is considered
  reasonably easy for bacteria, but is very difficult for eukaryotes.
  <LI> Gene identification through alignment with know proteins and
  EST sequences (Expressed Sequence Tags; mRNA sequences).
  <LI> Gene prediction through similarity with proteins or ESTs
  in other organisms.
  <LI> Gene prediction through comparison with other genomes;
  conserved regions are probably coding or regulatory regions. This is
  called synteny, and is very promising for analysis of higher
  eukaryote genomes.
  </UL>
<P>
<LI> Annotation of the genes: Compare with genes/proteinsof known
function in other organisms. This is essentially the same as labelling
the gene.
<P>
<LI> Functional classification. Broad groups of functional
characterization, such as 'ribosomal proteins', 'nucleotide
metabolism', 'signal transduction'.
<P>
<LI> Metabolic pathways.
  <UL>
  <LI> Are any common pathways missing?
  <LI> Are there 'gaps' (missing enzymes) in some pathways?
  <LI> Compare identified pathways with the life style of the
  organism.
  </UL>
<P>
<LI> Evolutionary history
  <UL>
  <LI> Internal genome duplications can sometimes be detected.
  <LI> Gene decay can sometimes be characterized: genes that are on
  their 'way out' after duplication, or because the life style of the
  organism has changed.
  <LI> Horisontal gene transfer: genes that have been acquired from
  another organism.
  </UL>
</UL>

<HR>
Copyright &copy; 2001
Per Kraulis
$Date: 2001/01/26 15:00:04 $
</BODY>
</HTML>
