<HTML>
<HEAD>
<TITLE>Lecture notes: 5. Protein sequence databases</TITLE>
</HEAD>
<BODY BGCOLOR="WHITE">

<A HREF="http://www.sbc.su.se/">Stockholm Bioinformatics Center, SBC</A>
<BR>
<A HREF="../index.html">Lecture notes, main page</A>

<P ALIGN="RIGHT">
Lecture 30 Oct 2001
Per Kraulis

</P>

<H2 ALIGN="CENTER">
<A HREF="index.html">Databases in bioinformatics</A>
</H2>

<H1 ALIGN="CENTER">5. Protein sequence databases</H1>

<P>
The two protein sequence databases SWISS-PROT and PIR are different
from the nucleotide databases in that they are both
<STRONG>curated</STRONG>. This means that groups of designated
curators (scientists) prepare the entries from literature and/or
contacts with external experts.
</P>

<H3>SWISS-PROT, TrEMBL
<A HREF="http://www.expasy.ch/sprot/">www.expasy.ch/sprot/</A>
</H3>

<P>
SWISS-PROT is a protein sequence database which strives to provide a
<STRONG>high level of annotations</STRONG> (such as the description of
the  function of a protein, its domains structure, post-translational
modifications, variants, etc.), a minimal level of redundancy and high
level of integration with other databases.
</P>

<P>
It was started in 1986 by Amos Bairoch in the Department of Medical
Biochemistry at the University of Geneva. This database is generally
considered one of the best protein sequence databases in terms of the
quality of the annotation.  Its size is given in the table below.
</P>

<P>
TrEMBL is a <STRONG>computer-annotated supplement</STRONG> of
SWISS-PROT that contains all the translations of EMBL nucleotide
sequence entries not yet integrated in SWISS-PROT. The procedure that
is used to produce it was developed by Rolf Apweiler. The annotation
of an entry in TrEMBL has not (yet) reached the standards required for
inclusion into SWISS-PROT proper. Its size is given in the table below.
</P>

<table border=1 cellpadding=4>
<TR>
<TH></TH>
<TH colspan=2>SWISS-PROT</TH>
<TH colspan=2>TrEMBL</TH>
</TR>
<TR>
<TH>Date</TH>
<TH>Release</TH>
<TH># entries</TH>
<TH>Release</TH>
<TH># entries</TH>
</TR>
<TR>
<TD>24 Oct 2001</TD>
<TD>40.1</TD>
<TD>101,737</TD>
<TD>18.0</TD>
<TD>484,388</TD>
</TR>
<TR>
<TD>2 Oct 2000</TD>
<TD>39.7</TD>
<TD>88,757</TD>
<TD>14.17</TD>
<TD>300,152</TD>
</TR>
</table>

<P>
SWISS-PROT and TrEMBL are developed by the SWISS-PROT groups at 
<A HREF="http://www.isb-sib.ch/">Swiss Institute of Bioinformatics
(SIB)</A> and at <A HREF="http://www.ebi.ac.uk">EBI</A>.
The databases can be accessed and searched through the 
<A HREF="http://www.expasy.ch/srs5/">the SRS system at ExPASy</A>, or
one can download the entire database as one single flat file. An
example of what an entry looks like is given for the 
<A HREF="kraf_human.sprot">human raf oncogene protein, ID
KRAF_HUMAN</A>.
</P>

<P>
The SWISS-PROT database has <STRONG>some legal restrictions</STRONG>:
the entries themselves are copyrighted, but freely accessible and
usable by academic researchers. Commercial companies must buy a
license fee from SIB.
</P>

<H3>PIR
<A HREF="http://pir.georgetown.edu/">pir.georgetown.edu</A>
</H3>

<P>
The Protein Information Resource (PIR) is a division of the National
Biomedical Research Foundation (NBRF) in the US. It is involved in a
collaboration with the
<A HREF="http://mips.gsf.de/">Munich Information Center
for Protein Sequences (MIPS)</A> and the Japanese International
Protein Sequence Database (JIPID). The PIR-PSD (Protein Sequence
Database) release 70.01 (22 Oct 2000) contains 254,293 entries.
</P>

<P>
PIR grew out of Margaret Dayhoff's work in the middle of the 1960s. It
strives to be <STRONG>comprehensive</STRONG>, well-organized,
accurate, and consistently annotated. However, it is generally
believed that it does not reach the level of completeness in the entry
annotation as does SWISS-PROT. Although SWISS-PROT and PIR overlap 
extensively, there are still many sequences which can be found in only
one of them.
</P>

<P>
One can search for entries or do sequence similarity searches at the
PIR site. The database can also be downloaded as a set of falt
files. An example of what an entry looks like is given for the
<A HREF="tvhuf6.pir">human raf-1 oncogene protein, ID TVHUF6</A>.
</P>

<P>
PIR also produces the <STRONG>NRL-3D</STRONG>, which is a database of
sequences extracted from the three-dimensional structures in
<A HREF="http://www.rcsb.org/pdb/">the Protein Databank (PDB)</A> (see
also <A HREF="3d.html">the following page in this
lecture</A>. The NRL_3D database makes the sequence information in
PDB available for similarity searches and retrieval and provides
cross-reference information for use with the other PIR Protein
Sequence Databases. 
</P>

<P>
It appears that the PIR web site, and possibly also the underlying
database, has improved considerably since one year ago. This means
that if one is interested in protein sequences, there is now even more
reason to check out PIR; SWISS-PROT is not the only game in town.
</P>

<HR>
Copyright &copy; 2001
Per Kraulis
$Date: 2001/11/09 15:19:05 $
</BODY>
</HTML>
