<HTML>
<HEAD>
<TITLE>Lecture notes: 2. PROSITE patterns and rules</TITLE>
</HEAD>
<BODY BGCOLOR="WHITE">

<A HREF="http://www.sbc.su.se/">Stockholm Bioinformatics Center, SBC</A>
<BR>
<A HREF="../index.html">Lecture notes: Structural biochemistry and bioinformatics 2001</A>

<P ALIGN="RIGHT">
Lecture 13 Nov 2001,
Per Kraulis
</P>

<H2 ALIGN="CENTER">
<A HREF="index.html">Hidden Markov models</A>
</H2>

<H1 ALIGN="CENTER">2. PROSITE patterns and rules</H1>

<P>
The <A HREF="http://www.expasy.ch/prosite/">PROSITE</A> database <A
HREF="../databases/motif.html">has already been mentioned in this
course</A>. It was initiated and is maintained by Amos Bairoch and
colleagues, now at the <A HREF="http://www.isb-sib.ch/">Swiss
Institute of Bioinformatics</A>. It is based on the proteins sequences
in SWISS-PROT. It aims at <STRONG>describing characteristic patterns
for some domain families using regular expressions</STRONG>, and
contains about 1400 patterns, rules and profile/matrices. It is being
maintained, but it is fair to say that it has been superceded in
practical terms by other search methods and databases, such as
<A HREF="http://www.cgb.ki.se/Pfam">Pfam</A>
(<A HREF="../multali/db.html">mentioned before</A>, and
<A HREF="pfam.html">discussed later</A>).
</P>

<P>
PROSITE makes a distinction between patterns and
rules, which are both described by regular
expressions:
<UL>
<LI> A <STRONG>pattern</STRONG> is intended to capture the
characteristic fingerprint of a protein domain family.
<LI> A <STRONG>rule</STRONG>, on the other hand, is intended to
highlight features in a protein sequence that does not necessarily
have anything to do with a specific protein family. For example,
potential glycosylation sites and phosphorylation sites can be found
in many protein sequences, and have little to do with the family of a
protein.
</UL>
</P>

<P>
In this section we will just describe the notation for patterns and
rules in PROSITE. Patterns and rules are described using the same
notation. <STRONG>Unfortunately, the PROSITE notation for sequence
patterns is different from the UNIX-type regular
expressions</STRONG>. However, the concepts are the same, and it is
not so difficult to translate a PROSITE pattern into a UNIX-type
regular expression.
</P>

<P>
As an example, let us use the PROSITE pattern
<A HREF="http://www.expasy.ch/cgi-bin/nicesite.pl?PS00562">CBM_1
(formerly CBD_FUNGAL)</A>
(accession code PS00562). The preceding link shows a nicer view of the 
entry. Below is the original text entry as it is given in the
downloadable PROSITE data file.
</P>

<PRE>
ID   CBD_FUNGAL; PATTERN.
AC   PS00562;
DT   DEC-1991 (CREATED); NOV-1997 (DATA UPDATE); JUL-1998 (INFO UPDATE).
DE   Cellulose-binding domain, fungal type.
PA   C-G-G-x(4,7)-G-x(3)-C-x(5)-C-x(3,5)-[NHG]-x-[FYWM]-x(2)-Q-C.
NR   /RELEASE=38,80000;
NR   /TOTAL=21(18); /POSITIVE=21(18); /UNKNOWN=0(0); /FALSE_POS=0(0);
NR   /FALSE_NEG=1; /PARTIAL=0;
CC   /TAXO-RANGE=??E??; /MAX-REPEAT=4;
CC   /SITE=1,disulfide; /SITE=7,disulfide; /SITE=9,disulfide;
CC   /SITE=16,disulfide;
DR   Q00023, CEL1_AGABI, T; Q12714, GUN1_TRILO, T; P07981, GUN1_TRIRE, T; 
DR   P07982, GUN2_TRIRE, T; P43317, GUN5_TRIRE, T; P46236, GUNB_FUSOX, T; 
DR   P46239, GUNF_FUSOX, T; P45699, GUNK_FUSOX, T; P15828, GUX1_HUMGR, T; 
DR   Q06886, GUX1_PENJA, T; P13860, GUX1_PHACH, T; P00725, GUX1_TRIRE, T; 
DR   P19355, GUX1_TRIVI, T; Q92400, GUX2_AGABI, T; P07987, GUX2_TRIRE, T; 
DR   P49075, GUX3_AGABI, T; P46238, GUXC_FUSOX, T; P50272, PSBP_PORPU, T; 
DR   O59843, GUX1_ASPAC, N; 
DO   PDOC00486;
//
</PRE>

<P>
The central line is the PA line, which contains the pattern. Let us go
through this pattern step by step.
<PRE>
PA   C-G-G-x(4,7)-G-x(3)-C-x(5)-C-x(3,5)-[NHG]-x-[FYWM]-x(2)-Q-C.
</PRE>
Let us go through the elements in the pattern to see what they mean:
<UL>
<LI>
Each <STRONG>non-x letter defines one particular type of amino-acid
residue</STRONG> in that position in the pattern. Here, we must have a
tripeptide Cys-Gly-Gly in the beginning of the matching segment of a
protein chain. The dash characters '-' add no information to the
pattern, and are added to make the pattern slightly easier to read.
<LI>
The notation <STRONG>x(4,7)</STRONG> means that at least 4 and at most
7 residues of any type may occur at this position. This corresponds to
the notation <STRONG>.{4,7}</STRONG> in a UNIX-type regular expression.
<LI>
The notation <STRONG>[NHG]</STRONG> means the same thing as in a
UNIX-type regular expression: in this position any of the residues
within the brackets may be chosen. One and only one such residue must
be at this position.
<LI>
The notation <STRONG>x(2)</STRONG> means that exactly two residues of
any type may occur at this position. This corresponds to the notation
<STRONG>..</STRONG> or <STRONG>.{2,2}</STRONG>in a UNIX-type regular
expression.
<LI>
The notation <STRONG>{GP}</STRONG> (not shown in this example) means
that all residues <STRONG>except</STRONG> Gly and Pro are allowed in
this position.
</UL>
</P>

<P>
The lines marked DR are the protein sequence entries in SWISS-PROT
that match (character T) or do not match (character N) the regular
expression. In this case, the protein GUX1_ASPAC does not match the
PROSITE rule, although it should; it is a false negative.
</P>

<HR>
Copyright &copy; 2001
Per Kraulis
$Date: 2001/11/09 15:30:22 $
</BODY>
</HTML>
