<HTML>
<HEAD>
<TITLE>Lecture notes: 8. Other relevant databases</TITLE>
</HEAD>
<BODY BGCOLOR="WHITE">

<A HREF="http://www.sbc.su.se/">Stockholm Bioinformatics Center, SBC</A>
<BR>
<A HREF="../index.html">Lecture notes, main page</A>

<P ALIGN="RIGHT">
Lecture 30 Oct 2001
Per Kraulis

</P>

<H2 ALIGN="CENTER">
<A HREF="index.html">Databases in bioinformatics</A>
</H2>

<H1 ALIGN="CENTER">8. Other relevant databases</H1>

<H3>GeneCards
<A HREF="http://www.genecards.org/index.shtml">www.genecards.org</A>
</H3>

<P>
GeneCards is a database of human genes, their products and their
involvement in diseases. It offers concise information about the
functions of all human genes that have an approved symbol, as well as
selected others. It is a typical example of a
<STRONG>secondary</STRONG> database, which contains many links to
other databases, and attempts to consolidate the information that is
available for a specific class of entity, in this case human genes.
</P>

<H3>GeneLynx
<A HREF="http://www.genelynx.org">www.genelynx.org</A>
</H3>

<P>
GeneLynx is a database of Web links for human genes. It contains
pointers to a large number of other databases. This is also a typical
secondary database. It is maintained by Boris Lenhard and Wyeth
Wasserman at CGB, KI, Sweden.
</P>

<H3>KEGG
<A HREF="http://www.genome.ad.jp/kegg/">www.genome.ad.jp/kegg/</A>
</H3>

<P>
The Kyoto Encyclopedia of Genes and Genomes (KEGG) is an effort to
computerize current knowledge of molecular and cellular biology in
terms of the information pathways that consist of interacting
molecules or genes and to provide links from the gene catalogs
produced by genome sequencing projects.
</P>

<H3>Amos' WWW links page
<A HREF="http://www.expasy.org/links.html">www.expasy.org/links.html</A>
</H3>

<P>
A page of many links to biological databases and/or web sites
(formerly known as Amos' links page).
</P>

<HR>
Copyright &copy; 2001
Per Kraulis
$Date: 2001/11/09 15:19:05 $
</BODY>
</HTML>
