<HTML>
<HEAD>
<TITLE>Lecture notes: 10. Databases of multiple alignments;
domains</TITLE>
</HEAD>
<BODY BGCOLOR="WHITE">

<A HREF="http://www.sbc.su.se/">Stockholm Bioinformatics Center, SBC</A>
<BR>
<A HREF="index.html">Lecture notes: Structural biochemistry and
bioinformatics 2000</A>

<P ALIGN="RIGHT">
Lecture 19 Jan 2001
Per Kraulis
</P>

<H2 ALIGN="CENTER">
<A HREF="multali.html">Sequence alignment</A>
</H2>

<H1 ALIGN="CENTER">10. Databases of multiple alignments; domains</H1>

<P>
Very early in the days of protein sequence analysis, it was observed
that some protein sequences contained long segments that were very
similar to other proteins, while the rest of the sequence in that
protein had no detectable similarity. Today, we take more or less for
granted that proteins are composed of <STRONG>domains</STRONG>,
segments of sequence which have been joined together by genetic events
during evolution so that the new protein has a function that is based
on the activities of the domains it contains.
</P>

<P>
Often the domains detectable by sequence analysis correspond to
structural domains in the 3D structure as well. There are now many
well-documented cases where it has been shown that domains can exists
perfectly well in isolation, when excised from the original
protein. Surprisingly often, a domain can be expressed and folded all
on its own.
</P>

<P>
There are today several databases that keep track of which domains
have been discovered, which proteins are involved, and that store the 
multiple sequence alignments of the relevant segments of the protein
sequences. <A HREF="databases-motif.html">We have already
discussed</A> one such database,
<A HREF="http://www.cgr.ki.se/Pfam/">Pfam</A>. Also, several of the
primary sequence databases now contain information about the domains
in the sequence entries.
</P>

<P>
The idea behind <STRONG>Pfam</STRONG> is twofold:
<OL>
<LI>
Create and maintain good-quality multiple sequence alignments of
well-defined protein sequence domains from proteins in SWISS-PROT.
<LI>
Use these multiple alignments for creating so-called HMMs that can be
used in profile searches of sequence databases.
</OL>
</P>

<P>
The multiple alignment used to define a domain (protein family) in
Pfam are called the <STRONG>seed alignment</STRONG>. It is created by
a curator, or taken from the literature. It is used to generate a
profile HMM for identifying other sequences in the databases
(SWISS-PROT and TREMBL) that contain the domain. The search results
are inspected to decide which cutoff should be used for that
particular Pfam entry. The search hits are then aligned automatically
into a so-called <STRONG>full alignment</STRONG>. 
</P>

<P>
There are a number of other useful databases of multiple sequence
alignments, such as:
<UL>
<LI>
<A HREF="http://www.bioinf.man.ac.uk/dbbrowser/PRINTS/">PRINTS</A>,
multiple motifs consisting of ungapped, aligned segments of sequences,
which serve as fingerprints for a protein family. 
<LI> 
<A HREF="http://blocks.fhcrc.org/">BLOCKS</A>, multiple motifs of
ungapped, locally aligned segments created automatically.
</UL>
</P>

<P>
These databases allow analysis of new sequences in terms of
<STRONG>which domains can be detected in the sequence</STRONG>. This
is often more useful, and sometimes also more sensitive (although this
is somewhat controversial) than doing sequence-to-sequence
comparisons. For instance, if a new protein has a kinase domain, then
it is more helpful to use a domain database (with some appropriate
search software, such as HMMER for Pfam) to identify it directly in
the sequence. The alternative, using BLAST or FASTA to find similar 
sequences, would return thousands of sequences, and it would require
some work to sort out that this is because the query sequence contains
a very common kinase domain.
</P>

<HR>
Copyright &copy; 2001
Per Kraulis
$Date: 2001/01/16 16:30:09 $
</BODY>
</HTML>
