<HTML>
<HEAD>
<TITLE>Lecture notes: 5. Multiple sequence alignment: why?</TITLE>
</HEAD>
<BODY BGCOLOR="WHITE">

<A HREF="http://www.sbc.su.se/">Stockholm Bioinformatics Center, SBC</A>
<BR>
<A HREF="index.html">Lecture notes: Structural biochemistry and
bioinformatics 2000</A>

<P ALIGN="RIGHT">
Lecture 19 Jan 2001
Per Kraulis
</P>

<H2 ALIGN="CENTER">
<A HREF="multali.html">Sequence alignment</A>
</H2>

<H1 ALIGN="CENTER">5. Multiple sequence alignment: why?</H1>

<P>
Pairwise alignments are fundamental and useful, but there are some
problems with them. For instance, when using one of the popular
sequence searching programs (FASTA, BLAST) which perform pairwise
alignments to find similar sequences in a database, one very often
obtains many sequences that are significantly similar to the query
sequence. Comparing each and every sequence to every other may be
possible when one has just a few sequences, but it quickly becomes
impractical as the number of sequences increases.
</P>
<P>
What we need is multiple sequence alignment, where <STRONG>all similar 
sequences can be compared in one single figure or table</STRONG>. The
basic idea is that the sequences are aligned on top of each other, so
that a coordinate system is set up, where <STRONG>each row is the
sequence</STRONG> for one protein, and <STRONG>each column is the 
'same' position</STRONG> in each sequence. Each column corresponds to
a specific residue in the 'prototypical' protein.
</P>
<P>
As with pairwise alignment, there will be gaps in some sequences,
most often shown by the dash '-' or dot '.' character. Note that to
construct a multiple alignment, one may have to introduce gaps in
sequences at positions where there were no gaps in the corresponding
pairwise alignment. This means that multiple alignments typically
contain more gaps than any given pair of aligned sequences. 
</P>

<HR>
Copyright &copy; 2001
Per Kraulis
$Date: 2001/01/16 16:29:29 $
</BODY>
</HTML>
